include/irr/asset/IAsset.h

```diff
@@ -131,22 +138,36 @@ class IAsset : virtual public core::IReferenceCounted
     void setMetadata(core::smart_refctd_ptr<IAssetMetadata>&& _metadata) { m_metadata = std::move(_metadata); }
 
 protected:
-    bool isDummyObjectForCacheAliasing;
+    bool isDummyObjectForCacheAliasing; //!< A bool for setting whether Asset is in dummy state. @see convertToDummyObject()
+
     //! To be implemented by base classes, dummies must retain references to other assets
     //! but cleans up all other resources which are not assets.
+	/**
+		Converting to a GPU object is equivalent of signing an Asset as dummy object.
```

Comment:
Not necessairly true.

It may be desireable to have a cache of GPU objects associated with the CPU objects, while at the same time being able to get rid of the RAM overhead for storage of such a CPU object.

This function will get rid of as much memory as possible (including references to CPU objects it needs), the only thing that stays is just the pointer and the memory allocated for that pointer (whole class).

After this operation we can treat the asset contents as invalid garbage.

The default CPU to GPU object converter provided by the engine does this.

Reply:
That's why I described that Assets are needed as a key, so they cannot be removed entirely

Reply:
yes, but this can be read very wrong... this class method has nothing to do with CPU to GPU object conversion

Reply:
well, I will move it
but in that case, please change name of the function, that name doesn't reflect reality